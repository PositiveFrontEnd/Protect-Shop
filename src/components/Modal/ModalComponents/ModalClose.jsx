import React from "react"
import PropTypes from 'prop-types'

const ModalClose = ({ onclick }) => {
    return (
        <button type="button" onClick={onclick} className="header__button">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="black">
                <path fillRule="evenodd" clipRule="evenodd" d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z" fill="white" />
            </svg>
        </button>
    )
}

ModalClose.defaultProps = {
    onclick: () => { }
}

ModalClose.propTypes = {
    onclick: PropTypes.func
}


export default ModalClose