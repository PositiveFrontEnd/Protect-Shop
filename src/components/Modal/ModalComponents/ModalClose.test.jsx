import React from "react";
import { fireEvent, getByText, render, screen } from "@testing-library/react";
import ModalClose from "./ModalClose";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
describe("test Modal", () => {
    test("render Modal", () => {
        const mockStore = configureStore();
        const initialState = {

        }
        const handelClick = () => console.log("hello");
        const store = mockStore(initialState);
        const { asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ModalClose onClick={handelClick} />
                </BrowserRouter>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});