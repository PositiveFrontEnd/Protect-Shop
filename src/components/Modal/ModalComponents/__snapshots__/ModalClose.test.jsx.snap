// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Modal > render Modal 1`] = `
<DocumentFragment>
  <button
    class="header__button"
    type="button"
  >
    <svg
      fill="black"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
        fill="white"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;
